journal/202101.md

```diff
@@ -0,0 +1,376 @@
+
+- completely [reworked](https://github.com/decred/dcrd/pull/2518) block indexing and processing to use headers-first semantics and support out of order processing of block data
+- reworked UTXO handling to work on a [per-outpoint](https://github.com/decred/dcrd/pull/2540) basis instead of at a transaction level
+- [sync model](https://github.com/decred/dcrd/pull/2555) reworked to use header announcements. The small 180-byte [headers](https://devdocs.decred.org/developer-guides/block-header-specifications/) will be downloaded before any block data, and then only needed blocks will be downloaded. This optimizes traffic, removes the concept of "orphan blocks", and opens the door for future enhancements.
+- preparations for implementing UTXO cache
+
+Other changes:
+
+- finalized [DCP-0006](https://github.com/decred/dcps/blob/master/dcp-0006/dcp-0006.mediawiki) document describing technical details of the upcoming Decentralized Treasury consensus change
+- added 3 new HTTPS [seeders](https://github.com/decred/dcrd/pull/2532) to reduce dependency on decred.org domain
+- the deprecated version 1 cfilters were [removed](https://github.com/decred/dcrd/pull/2525) in favor of the more efficient v2 ("compact filters" power Decred's awesome [SPV](https://docs.decred.org/wallets/spv/) tech for light clients)
+- added TLS client [authentication](https://github.com/decred/dcrd/pull/2482) using client certificates (more flexible than user+password)
+- added [RSA](https://github.com/decred/dcrd/pull/2551) 4096 bit key support to gencerts tool
+- new commands to [invalidate](https://github.com/decred/dcrd/pull/2536) arbitrary blocks and reconsider arbitrary blocks for validation (allows easier testing and smoother upgrades in the future)
+- prevent [duplicate](https://github.com/decred/dcrd/pull/2563) pending connections to same IP address and port
+- miscellaneous updates to test code, logging, docs, dependencies, minor fixes, etc
+
+A total of 47 PRs from 10 contributors were [merged](https://github.com/decred/dcrd/pulls?q=is%3Apr+merged%3A2021-01-01..2021-01-31+sort%3Aupdated-asc), adding 12K and deleting 7K lines of code.
+
+In progress:
+
+- introduce [Age-Partitioned Bloom Filters](https://github.com/decred/dcrd/pull/2579) to significantly reduce memory used for tracking what data other peers are known to have
+
+**[dcrwallet](https://github.com/decred/dcrwallet)**
+
+- VSP client was [rewritten](https://github.com/decred/dcrwallet/pull/1956) to improve background processing and support recovery of fee payment information after seed restores { better translation to User-English is welcome, I didn't quite grasp the first design goal of 1956 }
+- narrow [rules](https://github.com/decred/dcrwallet/pull/1961) of what is considered a possible CoinJoin { I got what this change does but now where it is used }
```

Comment:
Used for automatically detecting a wallets mixed account during recovery from seed.

Reply:
Updated.